src/Eventso.Subscription/Observing/Batch/ErrorHandling/BreakdownFailedBufferProcessor.cs

```diff
@@ -0,0 +1,77 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Eventso.Subscription.Observing.Batch.ErrorHandling
+{
+    internal class BreakdownFailedBufferProcessor<TEvent> : IFailedBufferProcessor<TEvent>
+        where TEvent : IEvent
+    {
+        private readonly IEventHandler<TEvent> _handler;
+        private readonly IConsumer<TEvent> _consumer;
+
+        public BreakdownFailedBufferProcessor(IEventHandler<TEvent> handler, IConsumer<TEvent> consumer)
+        {
+            _handler = handler;
+            _consumer = consumer;
+        }
+
+        public async Task Process(
+            IReadOnlyList<Buffer<TEvent>.BufferedEvent> bufferedEvents,
+            CancellationToken cancellationToken)
+        {
+            SingleEventCollection convertibleCollection = null;
+
+            for (var i = 0; i < bufferedEvents.Count; i++)
+            {
+                var bufferedEvent = bufferedEvents[i];
+
+                if (bufferedEvent.Skipped)
+                    continue;
+
+                var @event = bufferedEvent.Event;
+
+                if (convertibleCollection is null)
+                    convertibleCollection = new SingleEventCollection(@event);
+                else
+                    convertibleCollection.SetEvent(@event);
+
+                await _handler.Handle(convertibleCollection, cancellationToken);
+
+                _consumer.Acknowledge(@event);
```

Comment:
nothing will acknowledge skipped events if retry handled successfully and skipped events were in the end of initial collection

Reply:
fair enough. seems like I should ack every single event, including skipped ones